apisix/plugins/redirect.lua

```diff
@@ -133,7 +135,11 @@ function _M.rewrite(conf, ctx)
         -- TODO： add test case
         -- PR: https://github.com/apache/apisix/pull/1958
         uri = "https://$host$request_uri"
-        ret_code = 301
+        if ngx.req.get_method() == http_post then
+            ret_code = 308
+        else
+            ret_code = 301
+        end
```

Comment:
User's configuration priority

`
if not ret_code or ret_code == 302 then
            if ngx.req.get_method() == "POST" then
                ret_code = 308
            else
                ret_code = 301
            end
        end
`

Reply:
I think when you choice http to https, you can not use user ret_code ,because http to https is an features, it means you set the redirect 301 but how handle post method , you can set 308, when the http is become https ,the user code is effect. what do you think @liuhengloveyou ?

Reply:
I think,,,user can choose not to configure it。If the user's configuration is wrong, should get the wrong result，
not be ignored。
The role of the user's configuration should be very definite.

Reply:
The http to https is a feature and the  ret_code is other feature, when apisix receive http request should change it to https, when apisix receive https we should redirect it to user ret_code, they are not conflict.